src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -371,17 +373,51 @@ private static AbstractTreeIterator prepareTreeParser(Repository repository, Obj
     public static boolean deleteDirectory(File path) {
         if (path.exists()) {
             File[] files = path.listFiles();
-            for (int i = 0; i < files.length; i++) {
-                if (files[i].isDirectory()) {
-                    deleteDirectory(files[i]);
-                } else {
-                    files[i].delete();
+            if (files != null) {
+                for (File file : files) {
+                    if (file.isDirectory()) {
+                        deleteDirectory(file);
+                    } else {
+                        boolean ignored = file.delete();
+                    }
                 }
             }
         }
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from remote origin HEAD, or from git config.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
+     */
+    public static String getDefaultBranchName(final String repositoryUrl) {
+        String branch;
+        try {
+            Ref head = Git.lsRemoteRepository().setRemote(repositoryUrl).callAsMap().get("HEAD");
```

Comment:
Here takes the way;

1. check remote default branch; as same as `git remote add origin $repositoryUrl; git remote show origin` and look into `HEAD branch: master`
2. when failed, then check local config; as same as `git config --global --get init.defaultBranch`
3. when there is no config, return "master"